src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -696,6 +790,27 @@ _state_need_kms_credentials (_state_machine_t *sm, bson_error_t *error)
    const bool callback_provided_aws =
       bson_iter_init_find (&iter, &creds, "aws");
 
+   // Whether the callback requested auto-Azure credentials
```

Comment:
May want to shift this below the `if (... && _needs_on_demand_aws_kms (...))` block to avoid breaking up the AWS-related code + add block scopes per provider.